src/write.js

```diff
@@ -56,7 +60,12 @@ function write(rows, geometry_type, geometries, callback) {
   shpView.setInt32(24, shpLength / 2);
   shxView.setInt32(24, (50 + geometries.length * 4));
 
-  var dbfBuf = dbf.structure(rows);
+  var dbfBuf;
```

Comment:
I might have just said `var dbfBuf  = attrCols ? dbf.structure(rows, attrCols) : dbf.structure(rows);` just to make it shorter but this is fine.